### PR TITLE
[AMDGPU] Error on non-global pointer with s_prefetch_data

### PR DESCRIPTION
**Pull request by rampitec**

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/SIISelLowering.cpp [hunk 1]
```diff
@@ -9929,9 +9929,9 @@ SDValue SITargetLowering::LowerINTRINSIC_VOID(SDValue Op,
     return SDValue(NewMI, 0);
   }
   case Intrinsic::amdgcn_s_prefetch_data: {
-    // For non-global address space preserve the chain and remove the call.
     if (!AMDGPU::isFlatGlobalAddrSpace(cast<MemSDNode>(Op)->getAddressSpace()))
-      return Op.getOperand(0);
+      report_fatal_error("s_prefetch_data only supports global or constant"
+                         " memory");
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SIISelLowering.cpp:9934]: Can check this in the IR verifier

Comment (rampitec) [reply llvm/lib/Target/AMDGPU/SIISelLowering.cpp]: Done.

File: llvm/lib/IR/Verifier.cpp [hunk 2]
```diff
@@ -6281,6 +6281,12 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
           "Value for inactive lanes must be a VGPR function argument", &Call);
     break;
   }
+  case Intrinsic::amdgcn_s_prefetch_data: {
+    unsigned AS = Call.getArgOperand(0)->getType()->getPointerAddressSpace();
+    Check(AS == 0 || AS == 1 || AS == 4,
```
[the remark this answers is not in the document]
Comment (rampitec) [reply llvm/lib/IR/Verifier.cpp]: Done

File: llvm/lib/IR/Verifier.cpp [hunk 3]
```diff
@@ -6281,6 +6281,12 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
...
+          "s_prefetch_data only supports global or constant memory");
```
[the remark this answers is not in the document]
Comment (rampitec) [reply llvm/lib/IR/Verifier.cpp]: Done

**Review by arsenm — APPROVED** [review]

**Pull request merged**